DSCResources/MSFT_xADDomainController/MSFT_xADDomainController.psm1

```diff
@@ -11,25 +11,42 @@ Import-Module -Name $adCommonFunctions
 
 function Get-TargetResource
 {
+    [CmdletBinding()]
     [OutputType([System.Collections.Hashtable])]
     param
     (
-        [Parameter(Mandatory)]
+        [Parameter(Mandatory = $true)]
+        [System.String]
         [String]$DomainName,
 
-        [Parameter(Mandatory)]
-        [PSCredential]$DomainAdministratorCredential,
+        [Parameter(Mandatory = $true)]
+        [System.Management.Automation.PSCredential]
+        $DomainAdministratorCredential,
 
-        [Parameter(Mandatory)]
-        [PSCredential]$SafemodeAdministratorPassword,
+        [Parameter(Mandatory = $true)]
+        [System.Management.Automation.PSCredential]
+        $SafemodeAdministratorPassword,
 
-        [String]$DatabasePath,
+        [Parameter()]
+        [System.String]
+        $DatabasePath,
 
-        [String]$LogPath,
+        [Parameter()]
+        [System.String]
+        $LogPath,
 
-        [String]$SysvolPath,
+        [Parameter()]
+        [System.String]
+        $SysvolPath,
 
-        [String]$SiteName
+        [Parameter()]
+        [System.String]
+        $SiteName,
+
+        [Parameter()]
+        [ValidateScript( { Test-Path $_} )]
+        [System.String]
+        $InstallationMediaPath
```

Comment:
On Get-TargetResource we don't need to add unusued non-mandatory parameters to the function

Reply:
I removed the parameter here but I don't believe there is a way to return this property because it is only used to install the database. I digress, when I removed it from Get I was getting an error,
![image](https://user-images.githubusercontent.com/8922086/51257597-f394e400-1975-11e9-8d2f-0e3dce7489a7.png)

I added back the parameter and the config applied.

Reply:
That error is happening because there are multiple places in the resource that splat @PsBoundParameters to Get-TargetResource.  If the function calling Get-TargetResource has more parameters, then you would have to remove them from PsBoundParameters first.

If we do include InstallationMediaPath in the Get result, then we would have to get the value from the parameter as-is.

Reply:
So I can remove it from @PsBoundParameters in test but I can't in the Set because it uses it, unless I am missing on how to do it. 

I am seeing if it logs using IFM for the install so it could be something I can do in Get but so far no luck.

Reply:
@regedit32  I did find that the DCPromo process logs to C:\Windows\debug\dcpromoui.log that records the IFM path. I could add this in the Get but I am struggling with the benefit that it would provide and if that log gets deleted then it would get $null from it. 

Thoughts?

Reply:
Similar to the `$SourcePath` property in [SqlSetup](https://github.com/PowerShell/SqlServerDsc/blob/e8e14af936e19951549559699d3214f192abe519/DSCResources/MSFT_SqlSetup/MSFT_SqlSetup.psm1#L485) resource, it is acceptable to just return the parameter value that is passed to it as-is.

Reply:
Ah.... I believe I was able to remove it from PsBoundParameters and Take it off Set.